### DISCUSSION TITLE
Gosund EP2 > Information: no power monitoring data

### DISCUSSION THREAD

**Discussion by cmonty14** [arendst/Tasmota#22171] [Support request]
Hi,
I'm running _Tasmota 14.2.0.4 (5d0cc81-lite)_ on this power plug: [Gosund SP111](https://templates.blakadder.com/gosund_EP2.html)
Generally the plug is working correctly.
However I don't get any power monitoring data in MQTT.
`19:45:31.247 MQT: tasmota/plug/gosund-ep2/dev4/STATE = {"Time":"2024-09-19T19:45:31","Uptime":"0T00:00:13","UptimeSec":13,"Vcc":3.478,"Heap":30,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":31,"MqttCount":1,"POWER":"ON","Wifi":{"AP":1,"SSId":"wifi-name","BSSId":"0A:55:31:xx:xx:xx","Channel":6,"Mode":"11n","RSSI":60,"Signal":-70,"LinkCount":1,"Downtime":"0T00:00:05"}}`

Can you pleaes advise how to fix this?

THX

**Comment by sfromis** [comment]
Since you mention both SP111 and EP2, not clear what you specifically mean, but `tasmota-lite` does not include energy monitoring support, you need to upgrade to full Tasmota if you want that.

And make sure to use the template matching your specific hardware variant.
https://templates.blakadder.com/gosund_SP111_v1_1.html
https://templates.blakadder.com/gosund_SP111_v1_4.html
https://templates.blakadder.com/gosund_EP2.html

(First and last of the 3 templates have the same energy monitoring pins defined)

**Comment by cmonty14** [comment]
Thanks for sharing this clarification.
Actually I have this power plug: _Gosund SP111 max. 3450W_

Unfortunately I don't know if this device matches to template v1.1 or v1.4; could you please advise?
I didn't open the plug and bought it /w _Tasmota firmware_ flashed already.

And my assumption is that I must install firmware `tasmota-lite` due to memory size restrictions. Is this correct?

**Reply by sfromis** [reply]
No, that was not a correct assumption. There are no ESP8266 devices where you could run `tasmota-lite` without having memory for full Tasmota; the minimum requirement is 1 MB of flash for all ESP8266 builds. The large majority of ESP8266-based IoT devices have this config.

Simplest way of checking for the two template versions would be to just try both. If you get working energy monitoring, it was the right one - assuming that you first make sure to have full Tasmota flashed.
